tests/cupy_tests/sparse_tests/test_coo.py

```diff
@@ -0,0 +1,234 @@
+import unittest
+
+import numpy
+try:
+    import scipy.sparse
+    scipy_available = True
+except ImportError:
+    scipy_available = False
+
+import cupy
+import cupy.sparse
+from cupy import testing
+
+
+def _make(xp, sp, dtype):
+    data = xp.array([0, 1, 2, 3], dtype)
+    row = xp.array([0, 0, 1, 2], 'i')
+    col = xp.array([0, 1, 3, 2], 'i')
+    # 0, 1, 0, 0
+    # 0, 0, 0, 2
+    # 0, 0, 3, 0
+    return sp.coo_matrix((data, (row, col)), shape=(3, 4))
+
+
+def _make_unordered(xp, sp, dtype):
+    data = xp.array([1, 4, 3, 2], dtype)
+    row = xp.array([0, 2, 1, 0], 'i')
+    col = xp.array([0, 2, 3, 1], 'i')
+    # 1, 2, 0, 0
+    # 0, 0, 0, 3
+    # 0, 0, 4, 0
+    return sp.coo_matrix((data, (row, col)), shape=(3, 4))
+
+
+@testing.parameterize(*testing.product({
+    'dtype': [numpy.float32, numpy.float64],
+}))
+class TestCooMatrix(unittest.TestCase):
+
+    def setUp(self):
+        self.m = _make(cupy, cupy.sparse, self.dtype)
+
+    def test_shape(self):
+        self.assertEqual(self.m.shape, (3, 4))
+
+    def test_ndim(self):
+        self.assertEqual(self.m.ndim, 2)
+
+    def test_nnz(self):
+        self.assertEqual(self.m.nnz, 4)
+
+    @unittest.skipUnless(scipy_available, 'requires scipy')
+    def test_get(self):
+        m = self.m.get()
+        self.assertIsInstance(m, scipy.sparse.coo_matrix)
+        expect = [
+            [0, 1, 0, 0],
+            [0, 0, 0, 2],
+            [0, 0, 3, 0]
+        ]
+        numpy.testing.assert_allclose(m.toarray(), expect)
+
+    @unittest.skipUnless(scipy_available, 'requires scipy')
+    def test_str(self):
+        self.assertEqual(str(self.m), '''  (0, 0)\t0.0
+  (0, 1)\t1.0
+  (1, 3)\t2.0
+  (2, 2)\t3.0''')
+
+    def test_toarray(self):
+        m = self.m.toarray()
+        expect = [
+            [0, 1, 0, 0],
+            [0, 0, 0, 2],
+            [0, 0, 3, 0]
+        ]
+        cupy.testing.assert_allclose(m, expect)
+
+
+@testing.parameterize(*testing.product({
+    'dtype': [numpy.float32, numpy.float64],
+}))
+@unittest.skipUnless(scipy_available, 'requires scipy')
+class TestCooMatrixInvalidInit(unittest.TestCase):
```

Comment:
Would you add tests for the case that row index and col index exceed matrix dimensions?